### DISCUSSION TITLE
Conditional Side Effect execution

### DISCUSSION THREAD

**Discussion by busslina** [GregoryConrad/rearch-dart#55] [Q&A] [answered]
Inside a `RearchConsumer`, can I execute a side effect conditionally based on a bool final field?

I ask this because I read somewhere I cannot do conditional execution inside a capsule

**Comment by GregoryConrad** [comment, marked as answer]
No, side effects must always be called or not called at all. The issue is if you call a side effect on one build and then don’t on the next, or vice versa.

**Reply by busslina** [reply]
So, for reassuring, this is correct:
```dart
class MyWidget extends RearchConsumer {
  final bool listenRouteChanges;

  const MyWidget({
    super.key,
    required this.listenRouteChanges,
  });

  @override
  Widget build(BuildContext context, WidgetHandle use) {
    if (listenRouteChanges) {
      use.initRouteChangeListener();
    }

    return Container();
  }
}
```

(*) I'm able to corrupt it if I have a subclass and I override `listenRouteChanges` with non consistent values (like `Random().nextBool()`), but it's up to me to do it well.

**Reply by GregoryConrad** [reply]
I would be *very* careful when doing that, and I probably wouldn’t recommend it, but yes that will work as anticipated.

**Reply by busslina** [reply]
Can you think of a workaround? Maybe to save the bool var initial value with `use.value`?

**Reply by GregoryConrad** [reply]
I’d use “final class” here to ensure it won’t be overridden and also document that it can’t be overridden because of the side effect.

but regardless, this is an interesting issue to find yourself in. Is there some other solution? Is the side effect actually needed (i.e. for rebuilds or something)?

**Reply by busslina** [reply]
Thanks. Actually my real class (`Screen`) is abstract
